### ISSUE TITLE
Autogenerate Keys Finds/Produces Incorrect Duplicate Keys

### ISSUE THREAD

**Issue by jkulesza** [JabRef/jabref#2089]
JabRef 3.7dev--snapshot--2016-09-28--master--555944f
mac os x 10.11.6 x86_64 
Java 1.8.0_102

Steps to reproduce:

1. Select all entries in DB.
2. Autogenerate BibTeX keys.
3. Cleanup entries below.

67 out of 110 of my entries have modified keys which append "a" as though there are duplicates (note that there are not duplicate entries in my .bib file confirmed with Quality -> Find Duplicates).  The associated PDFs are accordingly renamed.  

If I Save/Exit/Re-launch and repeat the process, the "a"s are removed and the database is returned to the original state.  

If I repeat the process two times before saving, those entries have "b" appended.  An additional time and I have "c" and so forth.  The only way to revert to have no letter is to Save/Exit/Re-key/Cleanup.

**Bottom line**: there appears to be a flaw in how entries are stored / uniqueness checked as part of the re-key process.

My key generator and cleanup settings are below.

<img width="727" alt="screen shot 2016-09-28 at 15 12 52" src="https://cloud.githubusercontent.com/assets/9939059/18932512/0ec208c6-858e-11e6-9209-66a64d0dd008.png">

<img width="624" alt="screen shot 2016-09-28 at 15 11 52" src="https://cloud.githubusercontent.com/assets/9939059/18932516/128db7b6-858e-11e6-83bf-2b31fd20dcdf.png">

**Comment by jkulesza** [comment]
Has anyone been able to reproduce?

**Comment by bartsch-dev** [comment]
It suffices if you select a single entry and generate the BibTeX key multiple times; the BibTeX key will count up (..a, ..b, ...).

I'll look into it in the next few days.

**Comment by koppor** [comment]
Thank you for reporting this issue. We think, that is already fixed in our development version and consequently the change will be included in the next release.

We would like to ask you to use a development build from https://builds.jabref.org/master.

**Comment by jkulesza** [comment]
Thank you for the quick fix!  I downloaded the development build and can confirm that the issue is fixed.
